cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -616,7 +644,18 @@ protected void revoke(final EntityId entityId, final Role role, Set<Action> acti
         client.close();
       }
     } catch (Exception e) {
-      throw Throwables.propagate(e);
+      // map sentry exceptions to appropriate cdap-security exceptions
+      if (e instanceof SentryAccessDeniedException || e instanceof SentryGrantDeniedException) {
+        throw new UnauthorizedException(e.getMessage());
+      } else if (e instanceof SentryGroupNotFoundException || e instanceof SentryNoSuchObjectException) {
+        throw new NotFoundException(e.getMessage());
+      } else if (e instanceof SentryAlreadyExistsException) {
+        throw new AlreadyExistsException(e.getMessage());
+      } else if (e instanceof SentryUserException) {
+        throw new BadRequestException(e.getMessage());
+      } else {
+        throw e;
```

Comment:
Seems like methods calling this method are not really expecting the exception, but rather just propagating it using `throws Exception`. Would it better to throw as unchecked exception from here?